### ISSUE TITLE
initialize application form with resource attributes by default

### ISSUE THREAD

**Issue by hiattp** [lockstep/rails_new#105]
Whenever I override the initialize method in ApplicationController I use the resource attributes as the default `params` instead of `{}`. This provides more intuitive behavior because you can initialize the form object passing only the resource and the form object will behave (state-wise) as the object itself (default values are set to whatever the current resource values are). Otherwise you can pass a user into the form initializer and no params (e.g. on an edit action) and see no default values populating the form.

I suggest we do something more like this in the application form, but wanted to propose it as a question first in case I'm missing any obvious disadvantages here:

```ruby
def initialize(resource = self.class.resource_class.new, params = resource.attributes)
  ...
end

**Comment by janklimo** [comment]
It makes sense to me though I have no experience with this use case. @vvmarulin?

**Comment by hiattp** [comment]
Once we get to the next phase of vmx we should just audit the custom form objects and see how often this pattern is used, that can be a heuristic for whether or not this will be a sensible default 👍

**Comment by janklimo** [comment]
@hiattp any new thoughts on this given that we have more experience with the subject now?

**Comment by hiattp** [comment]
Yep, we actually made this change in VmX almost a year ago and it seems to be working well I think it is template-worthy at this point 👍
